src/index.tsx

```diff
@@ -34,12 +39,20 @@ Those are known to block WebAssembly.`);
     }
 }
 
-ReactDOM.render(
-    <div className="initial-load">
-        <div className="fa fa-spinner fa-spin"></div>
-        <div className="initial-load-text">Loading...</div>
-    </div>,
-    document.getElementById("base"),
-);
+async function loadReact() {
+    const React = await ReactImport;
+    const ReactDOM = await ReactDOMImport;
+
+    if (!uiLoaded) {
+        ReactDOM.render(
+            <div className="initial-load">
+                <div className="fa fa-spinner fa-spin"></div>
+                <div className="initial-load-text">Loading...</div>
+            </div>,
+            document.getElementById("base"),
+        );
+    }
+}
```

Comment:
You don't need this function. You can just move this div into the HTML template (src/index.html).

Reply:
When I move it into the HTML template, the CSS files that we import in `index.tsx` don't get applied to it, so the loading screen looks like this:

![chrome_J3GBhIiNLp](https://user-images.githubusercontent.com/8262173/72620881-ce7ade80-3905-11ea-9305-505cb3540aea.png)

Reply:
You likely kept `className`. You need to rename it to `class`. It's `className` in jsx / tsx because `class` is a keyword.

Reply:
oh oops, i'm dumb lol